### PR TITLE
Add Azure configuration block

### PR DESCRIPTION
**Pull request by rneacsu5**
This pull request adds the `azure_config` provider configuration block which allows the user to pass Azure credentials and other configuration options.

This is useful when relying on the `DefaultAzureCredential` feature of the Azure SDK is not sufficient, such as when configuring multiple servers across multiple tenants from the same Terraform state.

```hcl
# Configure the MySQL provider for Azure MySQL Server with specific credentials
provider "mysql" {
  endpoint = "azure://your-azure-instance-name.mysql.database.azure.com"
  username = "username@yourtenant.onmicrosoft.com"

  azure_config {
    tenant_id     = "your-tenant-id"
    client_id     = "your-client-id"
    client_secret = var.client_secret
  }
}
```

All variables are optional, falling back to their corresponding environment variables if absent. If the entire block is missing, the previous behaviour is preserved and `DefaultAzureCredential` is used.

Tested this manually. Unfortunately, I don't think this can be tested automatically without a test Azure account and an AAD enabled MySQL Server.

### CODE REVIEW

**Comment by kratkyzobak** [comment]
Hi @petoju, thanks for letting me know. 

tl;dr I'm OK with this changes.

My opinion as user - until there is fallback to `DefaultAzureCredential`, it is not breaking change, so "I don't care".

Opinion as developer (remains same as I stated in PR introducing Entra ID authentication support) - there should be tons of configurations to provide all possible variants of Azure login (as seen in https://github.com/hashicorp/terraform-provider-azuread/blob/main/internal/provider/provider.go#L98-L235). I did not tried to implement any of them in advance (to not need to test them) and instead of it wait for someone who would need them enough to implement them. So subset of these configurations are now implemented to provider, others are still accessible using `DefaultAzureCredential`.

**Review by petoju — APPROVED** [review]

**Comment by petoju** [comment]
Thanks for the evaluation! This was released as v3.0.59.

**Comment by rneacsu5** [comment]
Thank you all for reviewing and merging this change!

@petoju I'll try to remember to extract that code in a separate function on the next occasion 👍
@kratkyzobak Indeed the current Azure config options are far from complete, but more can be added as needed.
